src/rpc/service_eth.go

```diff
@@ -207,6 +218,17 @@ func (service *EthService) SendRawTransaction(ctx context.Context, s string) (*c
 	}
 	utils.Logger.Info().Hex("Incoming tx hash", txHash.Bytes()).Hex("Encrypted tx hash", submitTx.Hash().Bytes()).Msg("Transaction sent")
 
+	service.Processor.Db.InsertNewTx(db.TransactionDetails{
+		Address:         fromAddress.String(),
+		Nonce:           tx.Nonce(),
+		TxHash:          txHash.String(),
+		EncryptedTxHash: submitTx.Hash().String(),
+		SubmissionTime:  time.Now().Unix(),
+	})
+
+	_ctx, _ := context.WithTimeout(context.Background(), time.Duration(service.Config.DelayInSeconds)*10*time.Second)
```

Comment:
I think we should also have a different config parameter for this as it has a different semantics, and we might want to update it to a different value than 10*the delay in seconds. Perhaps also name it WaitMinedTimeout.